packages/fxa-auth-server/lib/routes/account.js

```diff
@@ -1512,19 +1512,48 @@ module.exports = (
 
         const { uid, scope } = request.auth.credentials;
         const { ecosystemAnonId } = request.payload;
-        const noneMatchHeader = request.headers['If-None-Match'];
         const scopeSet = ScopeSet.fromArray(scope);
 
+        function parseMultiValueHeader(named) {
+          const value = request.headers[named];
+          if (!value || !value.length) {
+            return [];
+          }
+          return value.toString().split(/,\s?/);
+        }
+
+        const ifNoneMatch = parseMultiValueHeader('If-None-Match');
+        const ifMatch = parseMultiValueHeader('If-Match');
+
         if (!scopeSet.contains('profile:ecosystem_anon_id:write')) {
           throw error.invalidScopes(scopeSet);
         }
 
         await customs.check(request, uid, 'updateEcosystemAnonId');
 
-        if (noneMatchHeader === '*') {
-          const account = await db.account();
-          if (account.ecosystemAnonId != null) {
-            throw error.anonIdExists();
+        if (ifNoneMatch.length || ifMatch.length) {
+          const { ecosystemAnonId: existingAnonId } = await db.account();
+
+          // Note the checks for both headers will accept a partial match
+          // So you can send as much or as little of the existing anon ID
+          // to compare against as you want.
+
+          if (existingAnonId != null && existingAnonId.length) {
+            if (ifNoneMatch.includes('*')) {
+              throw error.anonIdUpdateConflict('If-None-Match');
+            } else if (
+              ifNoneMatch.length &&
+              ifNoneMatch.some((value) => existingAnonId.includes(value))
```

Comment:
I thought we were going to do a proper content hash as the key/etag/whatever?

Reply:
😑 Yup. Slack is hard.